code/modules/mob/living/basic/space_fauna/carp/carp_ai_rift_actions.dm

```diff
@@ -201,7 +201,7 @@
 		if (!exit_count)
 			continue
 		var/turf/rift_exit = rift.exit_locs[exit_count]
-		if (get_dist(rift_exit, target) >= distance_to_target)
+		if (get_dist(rift_exit, target) >= distance_to_target || get_dist(rift, target) >= distance_to_target)
```

Comment:
```suggestion
		if ((get_dist(rift_exit, target) + get_dist(rift, target)) >= distance_to_target)
```

Otherwise you'll still have situations when you'll be wasting time